prov/gni/src/gnix_mr.c

```diff
@@ -65,10 +65,11 @@ void _gnix_convert_key_to_mhdl(
 		IN    gnix_mr_key_t *key,
 		INOUT gni_mem_handle_t *mhdl)
 {
-	uint64_t va = (uint64_t) __sign_extend(key->pfn << PAGE_SHIFT,
-			GNIX_MR_VA_BITS);
+	uint64_t va = key->pfn;
 	uint8_t flags = 0;
 
+	va = (uint64_t) __sign_extend(va << PAGE_SHIFT, GNIX_MR_VA_BITS);
+
 	if (key->flags & GNIX_MR_FLAG_READONLY)
```

Comment:
The pfn bitfield isn't wide enough to hold the shifted va value.  The result va was being truncated.